src/nccl_ofi_sendrecv.c

```diff
@@ -1782,11 +1779,14 @@ static inline int create_send_comm(nccl_net_ofi_conn_handle_t *handle,
 	if (OFI_UNLIKELY(ret != 0)) {
 		NCCL_OFI_WARN("Could not allocate NCCL OFI requests free list for dev %d",
 			      device->base.dev_id);
-		free(ret_s_comm);
-		return ret;
+		goto out;
 	}
 
 	*s_comm = ret_s_comm;
+out:
+	if (ret)
```

Comment:
`if (ret_s_comm)`?

Reply:
The happy path won't be happy if I free it always.  I only want to free it if fi_getname() or the freelist init failed (with non-zero ret).

Reply:
Oops, you're right 😊